cygwin: make math.log2() exact for powers of 2

The log2() from newlib is off by one ulp for some powers of 2.

--- Python-3.14.10.orig/Modules/mathmodule.c
+++ Python-3.14.10/Modules/mathmodule.c
@@ -712,6 +712,13 @@
     }
 
     if (x > 0.0) {
+#ifdef __CYGWIN__
+        /* The log2() from newlib isn't exact for powers of 2 */
+        int e;
+        if (frexp(x, &e) == 0.5) {
+            return e - 1.0;
+        }
+#endif
         return log2(x);
     }
     else if (x == 0.0) {
